launch/launch/substitutions/python_expression.py

```diff
@@ -47,26 +48,61 @@ def __init__(self, expression: SomeSubstitutionsType) -> None:
             'expression',
             'PythonExpression')
 
+        ensure_argument_type(
+            python_modules,
+            (str, Substitution, collections.abc.Iterable),
+            'python_modules',
+            'PythonExpression')
+
         from ..utilities import normalize_to_list_of_substitutions
         self.__expression = normalize_to_list_of_substitutions(expression)
+        self.__python_modules = normalize_to_list_of_substitutions(python_modules)
 
     @classmethod
     def parse(cls, data: Iterable[SomeSubstitutionsType]):
         """Parse `PythonExpression` substitution."""
-        if len(data) != 1:
-            raise TypeError('eval substitution expects 1 argument')
-        return cls, {'expression': data[0]}
+        if len(data) < 1 or len(data) > 2:
+            raise TypeError('eval substitution expects 1 or 2 arguments')
+        kwargs = {}
+        kwargs['expression'] = data[0]
+        if len(data) == 2:
+            # We get a text subsitution from XML,
+            # whose contents are comma-separated module names
+            kwargs['python_modules'] = []
+            # Check if we got empty list from XML
+            if len(data[1]) > 0:
+                modules_str = data[1][0].perform(None)
+                kwargs['python_modules'] = modules_str.split(', ')
```

Comment:
Might want to split by "," and then `strip()` each element (so it supports both "a, b, c" and "a,b,c")